src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/validation/function.ts

```diff
@@ -357,7 +394,34 @@ export function validateFunction({
       return messages;
     }
   }
+
   const matchingSignatures = getSignaturesWithMatchingArity(fnDefinition, fn);
+
+  // Check license requirements for the function
+  const hasMinimumLicenseRequired = context.hasMinimumLicenseRequired;
```

Comment:
You have it at the callbacks (which makes more sense than the context as it is a callback). It doesnt makie sense to have it in both places. You should use the existing one in callbacks.